src/Compilers/Core/Portable/DiagnosticAnalyzer/SuppressMessageAttributeState.cs

```diff
@@ -46,7 +46,7 @@ public void AddGlobalSymbolSuppression(ISymbol symbol, SuppressMessageInfo info)
                 }
                 else
                 {
-                    suppressions = new Dictionary<string, SuppressMessageInfo>() {{ info.Id, info }};
+                    suppressions = new Dictionary<string, SuppressMessageInfo>() { { info.Id, info } };
                     _globalSymbolSuppressions.Add(symbol, suppressions);
```

Comment:
Yes you are probably right. We should consider using a SmallDictionary or even just maintain a list of infos per symbol.